### PR TITLE
TS-3939: Add a --enable-asan configure option

### PR DESCRIPTION
**Pull request by bryancall**

### CODE REVIEW

**Comment by atsci** [comment]
FreeBSD build *successful*! See https://ci.trafficserver.apache.org/job/Github-FreeBSD/1128/ for details.

**Comment by zwoop** [comment]
Looks good to me. A future improvement would be that --enable-asan disables the freelist. It's today pretty obnoxious to run with ASAN, since you have to tell traffic_manager to pass along a command line option to traffic_server to disable the freelist.

File: configure.ac [hunk 1]
```diff
@@ -188,6 +188,14 @@ AC_ARG_ENABLE([werror],
 )
 AC_MSG_RESULT([$enable_werror])
 
+# Enable ASAN for the builds
+AC_MSG_CHECKING([whether to enable asan])
+AC_ARG_ENABLE([asan],
+  [AS_HELP_STRING([--enable-asan],[turn on asan])],
```
Comment (jpeach) [inline configure.ac]: "turn on Address Sanitizer"?

**Comment by atsci** [comment]
Linux build *successful*! See https://ci.trafficserver.apache.org/job/Github-Linux/1022/ for details.

**Pull request merged**